### PR TITLE
objectively terrible solution to the captain's gas mask scaring people with Greytider Phobia

### LINKED ISSUE #62452: Fear of greytide includes security and explorer gas masks
Reporting client version: 514.1569

## Round ID:
[172166](https://scrubby.melonmesa.com/round/172166)

## Testmerges:
- [Undoes "Removes CHECK_TICK from overlays, following the same theory as instant explosions" (#58382)](https://www.github.com/tgstation/tgstation/pull/61919)
- [Fixes halloween roundstart species](https://www.github.com/tgstation/tgstation/pull/62427)

## Reproduction:
Get fear of greytide phobia
Die because you get scared of a security gas mask

### PR DESCRIPTION
**Pull request by phasenoisepon**
## About The Pull Request
- PR'ed for Hacktoberfest (so if this passes muster I'd appreciate it being tagged with `hacktoberfest-accepted`)

closes https://github.com/tgstation/tgstation/issues/62452
## Why It's Good For The Game

People that are scared of assistants now won't be scared of Security/The Captain
Previously: https://streamable.com/vdej9j
Fixed: https://streamable.com/s4pzf9

## Changelog

:cl:
fix: Gas mask subtypes will no longer scare people with Greytider Phobia
/:cl:

### CODE REVIEW

File: code/_globalvars/traumas.dm [hunk 1]
```diff
@@ -126,10 +126,9 @@ GLOBAL_LIST_INIT(phobia_objs, list(
 		/obj/item/clothing/under/plasmaman/clown
 	)),
 
-	"greytide" = typecacheof(list(/obj/item/clothing/under/color/grey, /obj/item/melee/baton/security/cattleprod,
-		/obj/item/spear, /obj/item/clothing/mask/gas, /obj/item/toy/figure/assistant,
-		/obj/structure/statue/sandstone/assistant
-	)),
+	"greytide" = (typecacheof(list(/obj/item/clothing/under/color/grey, /obj/item/melee/baton/security/cattleprod,
+		/obj/item/spear, /obj/item/toy/figure/assistant,
+		/obj/structure/statue/sandstone/assistant))+typecacheof(list(/obj/item/clothing/mask/gas), FALSE, TRUE)),
```
Comment (Mothblocks) [inline code/_globalvars/traumas.dm:131]: This is unreadable, what does this do? What does `FALSE, TRUE` mean? Can you annotate them with named vars? 

Phobia code is so ugly to read, nothing split out.

Comment (phasenoisepon) [reply code/_globalvars/traumas.dm]: a6f2987 should resolve this?

**Comment by CRITAWAKETS** [comment]
captain's gas mask specifically should still trigger greytider phobia

captains are just bluer assistants

**Comment by SuperNovaa41** [comment]
> captain's gas mask specifically should still trigger greytider phobia
> 
> captains are just bluer assistants

yeah but this also fixes security masks and explorer masks causing problems

**Comment by MMMiracles** [comment]
i believe repos need to opt out of hacktoberfest for PRs not to count, so the tag probably isn't necessary. at least the ones i've submitted have been seen by their system without the tags.

File: code/_globalvars/traumas.dm [hunk 2]
```diff
@@ -126,9 +126,10 @@ GLOBAL_LIST_INIT(phobia_objs, list(
 		/obj/item/clothing/under/plasmaman/clown
 	)),
 
-	"greytide" = typecacheof(list(/obj/item/clothing/under/color/grey, /obj/item/melee/baton/security/cattleprod,
-		/obj/item/spear, /obj/item/clothing/mask/gas, /obj/item/toy/figure/assistant,
-		/obj/structure/statue/sandstone/assistant
+	"greytide" = (typecacheof(list(/obj/item/clothing/under/color/grey, /obj/item/melee/baton/security/cattleprod,
+		/obj/item/spear, /obj/item/toy/figure/assistant,
+		/obj/structure/statue/sandstone/assistant)) + typecacheof(list(/obj/item/clothing/mask/gas), ignore_root_path = FALSE, only_root_path = TRUE
```
Comment (Fikou) [inline code/_globalvars/traumas.dm:131]: why do you use typecache of at all, why not just add the path for the gas mask

Comment (phasenoisepon) [reply code/_globalvars/traumas.dm:131]: specific check for type in typecache in phobia.dm line 51:
```			if(is_type_in_typecache(O, trigger_objs))```

**Comment by optimumtact** [comment]
@MMMiracles the label makes it count without needing it to be merged.

**Comment by phasenoisepon** [comment]
> @MMMiracles the label makes it count without needing it to be merged.

don't tell them about my side hustle u hoe

**Comment by MMMiracles** [comment]
future lifehacks for the next hacktoberfest, thank you

**Comment by Mothblocks** [comment]
Phobia code is really shit, and so I'm going to hope that this really is the best way to do it without some major refactors to allow single types without entire typecaches (and without just doing `gasmask = TRUE` directly)

**Pull request merged**